[clang-tidy] Fix modernize-use-integer-sign-comparison comparison

qt-tatiana opened the pull request:
- modernize-use-integer-sign-comparison should ignore a comparison between the signed wide type and the unsigned narrow type, see #120867

qt-tatiana committed:
[clang-tidy] Fix modernize-use-integer-sign-comparison comparison

- modernize-use-integer-sign-comparison should ignore a comparison between
the signed wide type and the unsigned narrow type, see #120867

PiotrZSL reviewed (commented):
Missing:
- Documentation entry to tell user about this behavior (single sentence)

To be considered:
- Configuration option to enable/disable this behavior

HerrCai0907 reviewed (commented):
Please update release note.
I agree to enable this feature by default if adding an option.

qt-tatiana committed:
[clang-tidy] Fix modernize-use-integer-sign-comparison comparison

- add ``ConsideringIntegerSize`` option, that ignores a comparison
between a signed wide and an unsigned narrow integer types, add
documentation.

qt-tatiana commented:
> Missing:
> 
> * Documentation entry to tell user about this behavior (single sentence)
> 
> To be considered:
> 
> * Configuration option to enable/disable this behavior

done

PiotrZSL commented on clang-tools-extra/docs/ReleaseNotes.rst:
```diff
@@ -301,6 +301,11 @@ Changes in existing checks
 
+- Improved :doc:`modernize-use-integer-sign-comparison
+  <clang-tidy/checks/modernize/use-integer-sign-comparison>` check to
+  add an option ``ConsideringIntegerSize``, that ignores a comparison between
```
Consider something in this style:
Improved XYZ check to ignore comparisons between signed wide and unsigned narrow integer types, configurable via the `ConsideringIntegerSize` option.

PiotrZSL commented on clang-tools-extra/docs/clang-tidy/checks/modernize/use-integer-sign-comparison.rst:
```diff
@@ -34,3 +34,7 @@ Options
   Default is `llvm`.
+
+.. option:: ConsideringIntegerSize
+  Ignores a comparison between a signed wide and an unsigned narrow
```
i think there should be empty line after option.

PiotrZSL reviewed (changes requested):
Few small nits, but functionally ok.

qt-tatiana committed:
[clang-tidy] Fix modernize-use-integer-sign-comparison comparison

- Delete unneccessary code, rename option to ``CheckIntegerSize``,
update ReleaseNotes.

qt-tatiana commented on clang-tools-extra/clang-tidy/modernize/ModernizeTidyModule.cpp:
```diff
@@ -126,6 +126,14 @@ class ModernizeModule : public ClangTidyModule {
+        ["modernize-use-integer-sign-comparison.ConsideringIntegerSize"] =
+        "true";
+    return Options;
+  }
```
done

qt-tatiana committed:
[clang-tidy] Fix modernize-use-integer-sign-comparison comparison

- fixed alignments

qt-tatiana commented:
Ping :)

qt-tatiana committed:
Merge branch 'main' into main
